InnerEye/ML/utils/io_util.py

```diff
@@ -412,24 +412,42 @@ def load_image_in_known_formats(file: Path,
         raise ValueError(f"Unsupported image file type for path {file}")
 
 
-def load_labels_from_dataset_source(dataset_source: PatientDatasetSource, check_exclusive: bool = True) -> np.ndarray:
+def load_labels_from_dataset_source(dataset_source: PatientDatasetSource, check_exclusive: bool = True,
+                                    image_size: Optional[Tuple[int]] = None) -> np.ndarray:
     """
     Load labels containing segmentation binary labels in one-hot-encoding.
     In the future, this function will be used to load global class and non-imaging information as well.
 
+    :type image_size: Image size, tuple of integers.
     :param dataset_source: The dataset source for which channels are to be loaded into memory.
-    :param check_exclusive: Check that the labels are mutually exclusive (defaults to True)
+    :param check_exclusive: Check that the labels are mutually exclusive (defaults to True).
     :return: A label sample object containing ground-truth information.
     """
-    labels = np.stack(
-        [load_image(gt, ImageDataType.SEGMENTATION.value).image for gt in dataset_source.ground_truth_channels])
 
-    if check_exclusive and (sum(labels) > 1.).any():  # type: ignore
+    if dataset_source.ground_truth_channels.count(None) > 0:
+        assert image_size is not None
+
+    label_list = []
+    # label_list keeps track of missing ground truth channels
+    for gt in dataset_source.ground_truth_channels:
+        if gt is None:
+            label_list.append(np.full(image_size, np.NAN, ImageDataType))
+        else:
+            label_list.append(load_image(gt, ImageDataType.SEGMENTATION.value).image)
+    labels = np.stack(label_list)
+
+    # If ground truth image is nan, then will not be used to check check_exclusive
+    # Image is nan, if voxel at index [0, 0, 0] is NaN
+    not_nan_label_images = [labels[label_id] for label_id in range(labels.shape[0])
+                            if not np.isnan(labels[label_id][0, 0, 0])]
```

Comment:
You can re-use your "is_missing..." function here

Reply:
Good point. Done